django/db/backends/base/schema.py

```diff
@@ -1020,7 +1037,9 @@ def _model_indexes_sql(self, model):
             output.append(self._create_index_sql(model, fields, suffix="_idx"))
 
         for index in model._meta.indexes:
-            output.append(index.create_sql(model, self))
+            sql = index.create_sql(model, self)
+            if sql is not None:
+                output.append(sql)
```

Comment:
I would add a flag to `Index`, e.g. `is_functional` that could be used here together with `supports_expression_indexes` to skip such indexes, e.g.
```python
if not index.is_functional or self.connection.features.supports_expression_indexes:
    output.append(index.create_sql(model, self))
```
Also we should return `None` in `_create_index_sql()` and `_delete_index_sql` if `index.is_functional` and `self.connection.features.supports_expression_indexes`